gammapy/data/data_store.py

```diff
@@ -455,6 +457,336 @@ def check(self, checks="all"):
+        obs_pos = self.obs_table[obs_mask]["EVENTS_FILENAME"]
+        path = make_path(os.environ["GAMMAPY_DATA"] + "/" + obs_pos[0])
+        size = int(os.path.getsize(path) / 1000.0)
+
+        tab["dataproduct_type"] = observation.obs_info["EXTNAME"]
+
+        tab["calib_level"] = 2  # Look into the data
+        tab["target_name"] = observation.obs_info["OBJECT"]
+
+        tab["obs_id"] = str(observation.obs_info["OBS_ID"])
+
+        tab["obs_collection"] = "DL3"
+        tab["obs_publisher_did"] = obs_publisher_did
+        tab["access_url"] = access_url
+        tab["access_format"] = "application/fits"
+
+        tab["access_estsize"] = size
+        tab["s_ra"] = observation.get_pointing_icrs(observation.tmid).ra.value
+        tab["s_dec"] = observation.get_pointing_icrs(observation.tmid).dec.value
+
+        tab["s_fov"] = 10.0
+        #    tab["s_region"] = 'null'
+        #    tab["s_resolution"] = np.nan
+        #    tab["s_xel1"] = 'null'
+        #    tab["s_xel2"] = 'null'
+        #    tab["t_xel"] = 'null'
+        tab["t_min"] = observation.tstart.value
+        tab["t_max"] = observation.tstop.value
+        tab["t_exptime"] = observation.observation_live_time_duration.value
+        #    tab["t_resolution"] = np.nan
+        #    tab["em_xel"] = 'null'
+        tab["em_min"] = observation.events.energy.min().value
+        tab["em_max"] = observation.events.energy.max().value
+        #    tab["em_res_power"] = np.nan
+        #    tab["o_ucd"] = 'null'
+        #    tab["pol_xel"] = 'null'
+        tab["facility_name"] = observation.obs_info["TELESCOP"]
+        tab["instrument_name"] = observation.obs_info["TELLIST"]
+
+        return tab
+
+    def to_obscore_table(self, selected_obs=None, format=None, **kwargs):
+
+        """Generate the complete obscore Table by stacking length=1 Tables given by DataStore._obscore_row()
+
+        Parameters
+        ----------
+        selected_obs : list or array of Observation ID (int)
+            (default of ``None`` means ``no obaservation ``)
+            If not given, the obscore default table is returned.
+
+        format : Astropy Table format
+            Define the format for the output obscore table
+
+
+        **kwargs : `str` {obs_publisher_did, access_url}
+        Giving the values for is highly recommended.
+        If any of these are not given the corresponding obscore field is left empty and a warning is raised for each empty value.
+
+        Returns
+        -------
+        Astropy Table length = len(selected_obs)
+        """
+
+        tab_default = self._obscore_def()
+
+        if selected_obs is None:
+            obscore_tab = tab_default
+
+        else:
+            obscore_tab = Table()
+            for i in range(0, len(selected_obs)):
+                obscore_row = self._obscore_row(selected_obs[i], **kwargs)
+                obscore_tab = vstack([obscore_tab, obscore_row])
+
+        self._write_obscore_file(obscore_tab, format)
```

Comment:
Why write it on disk here?

Reply:
I removed for now the function to write the table, currently to_oscore_table returns an astropy table object.